Source/KeyboardManager.swift

```diff
@@ -5,35 +5,51 @@
 
 import UIKit
 
-public typealias KeyboardManagerEventClosure = (KeyboardManagerEvent, KeyboardManagerEventData) -> Void
+public typealias KeyboardManagerEventClosure = (KeyboardManagerEvent) -> Void
 
 public enum KeyboardManagerEvent {
-    case willShow
-    case didShow
-    case willHide
-    case didHide
-}
+    case willShow(KeyboardManagerEvent.Data)
+    case didShow(KeyboardManagerEvent.Data)
+    case willHide(KeyboardManagerEvent.Data)
+    case didHide(KeyboardManagerEvent.Data)
 
-public struct KeyboardManagerEventData {
     public struct Frame {
         var begin: CGRect
         var end: CGRect
     }
-    var frame: Frame
-    var animationCurve: String
-    var animationDuration: Double
-    var isLocal: Bool
-
-    static func null() -> KeyboardManagerEventData {
-        let frame = Frame(begin: CGRect.zero, end: CGRect.zero)
-        return KeyboardManagerEventData(frame: frame, animationCurve: "", animationDuration: 0.0, isLocal: false)
+
+    public struct Data {
+
+        var frame: Frame
+        var animationCurve: String
+        var animationDuration: Double
+        var isLocal: Bool
+
+        static func null() -> Data {
+            let frame = Frame(begin: CGRect.zero, end: CGRect.zero)
+            return Data(frame: frame, animationCurve: "", animationDuration: 0.0, isLocal: false)
+        }
+    }
+
+    var data: KeyboardManagerEvent.Data {
+        switch self {
+        case let .willShow(data):
+            return data
+        case let .didShow(data):
+            return data
+        case let .willHide(data):
+            return data
+        case let .didHide(data):
+            return data
+        }
     }
 }
 
 public protocol KeyboardManagerProtocol {
 
     var eventClosure: KeyboardManagerEventClosure? { get set }
 
+    func bindToKeyboardNotifications(scrollView: UIScrollView)
 }
 
 public final class KeyboardManager {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/alphatroya/KeyboardManager/blob/a51863b5a29b6326270bd93f1c4098a9bafc347f/Source/KeyboardManager.swift#L55"></span>Public declarations should be documented.</td>
    </tr>
  </tbody>
</table>